solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -196,6 +196,17 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         require(updatingLiquidity, "ERR_NOT_UPDATING_LIQUIDITY");
     }
 
+    // ensures that the portion is valid
+    modifier validPortion(uint32 _portion) {
+        _validPortion(_portion);
+        _;
+    }
+
+    // error message binary size optimization
+    function _validPortion(uint32 _portion) internal pure {
+        require(_portion > 0 && _portion <= PPM_RESOLUTION, "ERR_INVALID_PORTION");
```

Comment:
@yudilevi : I don't get it, this is exactly what I did here (except it is not exactly a "valid PPM" checker, since 0 is also valid).

Reply:
I guess we could use such a modifier in multiple places

Reply:
You're right, we actually do have something similar in a few places:
```
require(_maxSystemNetworkTokenRatio <= PPM_RESOLUTION, "ERR_INVALID_MAX_RATIO");
require(_averageRateMaxDeviation <= PPM_RESOLUTION, "ERR_INVALID_MAX_DEVIATION");
```
But these cases don't prohibit 0 as input, while the `validPortion` modifier does.
Also, we use a unique error-message in each one of them.
But each one of them is for a standalone AND owner-only function, which is invoked from a separate flow, so even if we use the same error-message, it will be easy to know the reason, because it's not like these functions will ever be executed from the same transaction.
So I say we go for a single modifier with a single error-message.
The only question is what to do about 0, because we don't allow portion 0, even though 0 is a valid PPM value.

Reply:
so we can combine these two but still leave 2 modifiers in place, one for > and the other for >=

Reply:
So you want the modifier to take min/max values? Just min value?
Also, what do you want the message to be? Again, keep in mind that 0 is a valid PPM.

Reply:
Fixed.